src/metabase/api/search.clj

```diff
@@ -477,26 +431,38 @@
 ;;; +----------------------------------------------------------------------------------------------------------------+
 
 (mu/defn ^:private search-context :- SearchContext
-  [search-string   :- [:maybe ms/NonBlankString]
-   archived-string :- [:maybe ms/BooleanString]
-   table-db-id     :- [:maybe ms/PositiveInt]
-   models          :- [:maybe [:or SearchableModel [:sequential SearchableModel]]]
-   limit           :- [:maybe ms/PositiveInt]
-   offset          :- [:maybe ms/IntGreaterThanOrEqualToZero]]
-  (cond-> {:search-string      search-string
-           :archived?          (Boolean/parseBoolean archived-string)
-           :current-user-perms @api/*current-user-permissions-set*}
-    (some? table-db-id) (assoc :table-db-id table-db-id)
-    (some? models)      (assoc :models
-                               (apply hash-set (if (vector? models) models [models])))
-    (some? limit)       (assoc :limit-int limit)
-    (some? offset)      (assoc :offset-int offset)))
+  [{:keys [archived-string
+           created-by
+           limit
+           models
+           offset
+           search-string
+           table-db-id]} :- [:map {:closed true}
+                             [:search-string                    [:maybe ms/NonBlankString]]
+                             [:archived-string {:optional true} [:maybe ms/BooleanString]]
+                             [:created-by      {:optional true} [:maybe ms/PositiveInt]]
+                             [:limit           {:optional true} [:maybe ms/Int]]
+                             [:models          {:optional true} [:maybe [:or SearchableModel [:sequential SearchableModel]]]]
+                             [:offset          {:optional true} [:maybe ms/Int]]
+                             [:table-db-id     {:optional true} [:maybe ms/PositiveInt]]]]
```

Comment:
Sorry to be nitpicky but can you order these too while you're at it?

Reply:
I chose to order this by : is_optional then alphabetical since whether sth is optional is more important imo.